### ISSUE TITLE
`where` is broken for lists of primitives

### ISSUE THREAD

**Issue by LeaVerou** [mavoweb/mavo#740]
Testcase by @karger: https://codepen.io/karger/pen/WNjbxVW?editors=1100

The non-rewritten form, `filter(entry, entry = 2)` works fine.

**Comment by DmitrySharabin** [comment]
I tried this code, and it worked. To be honest, I felt this more logical and decided to give it a try. And it turned out that Mavo already understands it.

`<button mv-action="delete(entry where 2)">delete 2</button>`

**Comment by karger** [comment]
This is actually more consistent with how I've seen "where" used with group objects, but I never would have thought of it (since it makes no sense in English).  And according to @LeaVerou both are supposed to work.

**Comment by DmitrySharabin** [comment]
If I translate `delete(entry where 2)` word-by-word from English into Russian it would perfectly make sense. Sometimes it’s useful to be a non-native among native speakers. 😅
Internationale teams rule the world.

**Comment by karger** [comment]
the problem is that there is no verb in the statement.  nothing that explains what relationship you want between entry and 2.  usually the condition includes a verb that returns a boolean.

**Comment by karger** [comment]
I've continued exploring and found more weirdness.  I've added to the codepen ( https://codepen.io/karger/pen/WNjbxVW?editors=1100 )  a "deletion by variable value" heading.   `delete(entry where 2)` works as discussed above.  But  `delete(entry where x)` when x has value 2 does *not* work the same way.

**Comment by LeaVerou** [comment]
@karger do note that from what I recall many people in the user study tried doing things like `entry where 2` before figuring out that it should be `entry where entry = 2`. 

I'm surprised that it works however, I'd expect it to match every item, as that's how operators work when one of the operand is a singleton.

**Comment by LeaVerou** [comment]
I think this is pretty low priority, and not on the critical path for v0.3.0. Bump.

**Comment by karger** [comment]
I agree that the bug is low priority but I wish I had a good understanding of what the exact semantics of "where" are.
